### DISCUSSION TITLE
how keda manager the hpa res which it created ？

### DISCUSSION THREAD

**Discussion by sunnoy** [kedacore/keda#1811] [Q&A / Need Help] [answered]
When I delete the hpa resource created by keda, it will be created automatically，

when i comment this line of code

https://github.com/kedacore/keda/blob/347e8975d5f7b7ccc8e2bfc15311c47fd267707d/controllers/scaledobject_controller.go#L106

 it will don‘t created automatically，How does keda implement this logic?

**Comment by zroubalik** [comment]
You should check Kubernetes Operator Pattern.

https://www.redhat.com/en/topics/containers/what-is-a-kubernetes-operator

**Reply by sunnoy** [reply]
https://github.com/kedacore/keda/blob/347e8975d5f7b7ccc8e2bfc15311c47fd267707d/controllers/hpa.go#L99

may be this

**Comment by sunnoy** [comment, marked as answer]
https://github.com/kedacore/keda/blob/347e8975d5f7b7ccc8e2bfc15311c47fd267707d/controllers/hpa.go#L99

**Reply by zroubalik** [reply]
that's just reference, the actual logic is in reconciler loop:
https://github.com/kedacore/keda/blob/347e8975d5f7b7ccc8e2bfc15311c47fd267707d/controllers/scaledobject_controller.go#L203-L226

**Reply by sunnoy** [reply]
thks
